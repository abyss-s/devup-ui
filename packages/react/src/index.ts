export { Box } from './components/Box'
export { Button } from './components/Button'
export { Center } from './components/Center'
export { Flex } from './components/Flex'
export { Grid } from './components/Grid'
export { Image } from './components/Image'
export { Input } from './components/Input'
export { Text } from './components/Text'
export { ThemeScript } from './components/ThemeScript'
export { VStack } from './components/VStack'
export { useTheme } from './hooks/use-theme'
export type { DevupProps } from './types/props'
export type { DevupTheme, DevupThemeColors } from './types/theme'
export type { DevupThemeTypography } from './types/typography'
export { css } from './utils/css'
export { getTheme } from './utils/get-theme'
export { globalCss } from './utils/global-css'
export { initTheme } from './utils/init-theme'
export { setTheme } from './utils/set-theme'
