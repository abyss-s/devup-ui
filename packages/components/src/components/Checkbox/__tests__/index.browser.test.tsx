import { fireEvent, render, screen } from '@testing-library/react'

import { Checkbox } from '..'

// Mock getTheme function
vi.mock('@devup-ui/react', async (importOriginal) => {
  const mod = await importOriginal<typeof import('@devup-ui/react')>()
  return {
    ...mod,
    getTheme: vi.fn(() => 'light'),
  }
})

const { getTheme } = await import('@devup-ui/react')

describe('Checkbox', () => {
  beforeEach(() => {
    // Reset to light theme before each test
    vi.mocked(getTheme).mockReturnValue('light')
  })

  afterEach(() => {
    vi.clearAllMocks()
  })
  it('should render', () => {
    const { container } = render(
      <Checkbox label="test-checkbox">Checkbox</Checkbox>,
    )
    expect(container).toMatchSnapshot()
  })

  it('should render with text children', () => {
    render(<Checkbox label="test-checkbox">Checkbox Label</Checkbox>)
    expect(screen.getByText('Checkbox Label')).toBeInTheDocument()
  })

  it('should render with custom React node children', () => {
    render(
      <Checkbox label="test-checkbox">
        <span data-testid="custom-label">Custom Label</span>
      </Checkbox>,
    )
    expect(screen.getByTestId('custom-label')).toBeInTheDocument()
  })

  it('should render checked state with CheckIcon', () => {
    const { container } = render(
      <Checkbox checked label="test-checkbox">
        Checked Checkbox
      </Checkbox>,
    )

    const checkIcon = container.querySelector('svg')
    expect(checkIcon).toBeInTheDocument()
    expect(checkIcon).toHaveAttribute('width', '12')
    expect(checkIcon).toHaveAttribute('height', '10')
  })

  it('should not render CheckIcon when unchecked', () => {
    const { container } = render(
      <Checkbox checked={false} label="test-checkbox">
        Unchecked Checkbox
      </Checkbox>,
    )

    const checkIcon = container.querySelector('svg')
    expect(checkIcon).not.toBeInTheDocument()
  })

  it('should handle disabled state', () => {
    render(
      <Checkbox disabled label="test-checkbox">
        Disabled Checkbox
      </Checkbox>,
    )

    const checkbox = screen.getByRole('checkbox')
    expect(checkbox).toBeDisabled()
  })

  it('should call onChange when clicked', () => {
    const handleChange = vi.fn()
    render(
      <Checkbox label="test-checkbox" onChange={handleChange}>
        Clickable Checkbox
      </Checkbox>,
    )

    const checkbox = screen.getByRole('checkbox')
    fireEvent.click(checkbox)

    expect(handleChange).toHaveBeenCalledWith(true)
  })

  it('should not call onChange when disabled and clicked', () => {
    const handleChange = vi.fn()
    render(
      <Checkbox disabled label="test-checkbox" onChange={handleChange}>
        Disabled Checkbox
      </Checkbox>,
    )

    const checkbox = screen.getByRole('checkbox')
    fireEvent.click(checkbox)

    expect(handleChange).not.toHaveBeenCalled()
  })

  it('should have proper label association', () => {
    render(<Checkbox label="test-checkbox">Test Label</Checkbox>)

    const checkbox = screen.getByRole('checkbox')
    const label = screen.getByText('Test Label')

    expect(checkbox).toHaveAttribute('id', 'test-checkbox')
    expect(label.closest('label')).toHaveAttribute('for', 'test-checkbox')
  })

  it('should render CheckIcon with disabled color when disabled and checked', () => {
    const { container } = render(
      <Checkbox checked disabled label="test-checkbox">
        Disabled Checked Checkbox
      </Checkbox>,
    )

    const checkIcon = container.querySelector('svg path')
    expect(checkIcon).toBeInTheDocument()
    expect(checkIcon).toHaveAttribute('fill', '#D6D7DE')
  })

  it('should render CheckIcon with normal color when enabled and checked', () => {
    const { container } = render(
      <Checkbox checked label="test-checkbox">
        Enabled Checked Checkbox
      </Checkbox>,
    )

    const checkIcon = container.querySelector('svg path')
    expect(checkIcon).toBeInTheDocument()
    expect(checkIcon).toHaveAttribute('fill', '#FFF')
  })

  describe('Theme-based styling', () => {
    it('should render CheckIcon with light theme disabled color when disabled and checked', () => {
      vi.mocked(getTheme).mockReturnValue('light')

      const { container } = render(
        <Checkbox checked disabled label="test-checkbox">
          Light Theme Disabled Checked
        </Checkbox>,
      )

      const checkIcon = container.querySelector('svg path')
      expect(checkIcon).toBeInTheDocument()
      expect(checkIcon).toHaveAttribute('fill', '#D6D7DE')
    })

    it('should render CheckIcon with dark theme disabled color when disabled and checked', () => {
      vi.mocked(getTheme).mockReturnValue('dark')

      const { container } = render(
        <Checkbox checked disabled label="test-checkbox">
          Dark Theme Disabled Checked
        </Checkbox>,
      )

      const checkIcon = container.querySelector('svg path')
      expect(checkIcon).toBeInTheDocument()
      expect(checkIcon).toHaveAttribute('fill', '#373737')
    })

    it('should render with light theme background colors', () => {
      vi.mocked(getTheme).mockReturnValue('light')

      const { container } = render(
        <Checkbox label="light-theme">Light Theme Checkbox</Checkbox>,
      )

      const checkbox = container.querySelector('input[type="checkbox"]')
      expect(checkbox).toBeInTheDocument()
      expect(getTheme).toHaveBeenCalled()
    })

    it('should render with dark theme background colors', () => {
      vi.mocked(getTheme).mockReturnValue('dark')

      const { container } = render(
        <Checkbox label="dark-theme">Dark Theme Checkbox</Checkbox>,
      )

      const checkbox = container.querySelector('input[type="checkbox"]')
      expect(checkbox).toBeInTheDocument()
      expect(getTheme).toHaveBeenCalled()
    })

    it('should use theme for disabled state background', () => {
      // Test light theme disabled
      vi.mocked(getTheme).mockReturnValue('light')

      const { container: lightContainer } = render(
        <Checkbox disabled label="light-disabled">
          Light Disabled
        </Checkbox>,
      )

      const lightCheckbox = lightContainer.querySelector(
        'input[type="checkbox"]',
      )
      expect(lightCheckbox).toBeInTheDocument()
      expect(getTheme).toHaveBeenCalled()

      // Test dark theme disabled
      vi.mocked(getTheme).mockReturnValue('dark')

      const { container: darkContainer } = render(
        <Checkbox disabled label="dark-disabled">
          Dark Disabled
        </Checkbox>,
      )

      const darkCheckbox = darkContainer.querySelector('input[type="checkbox"]')
      expect(darkCheckbox).toBeInTheDocument()
      expect(getTheme).toHaveBeenCalled()
    })

    it('should use theme for checked disabled state background', () => {
      // Test light theme checked disabled
      vi.mocked(getTheme).mockReturnValue('light')

      const { container: lightContainer } = render(
        <Checkbox checked disabled label="light-checked-disabled">
          Light Checked Disabled
        </Checkbox>,
      )

      const lightCheckbox = lightContainer.querySelector(
        'input[type="checkbox"]',
      )
      expect(lightCheckbox).toBeInTheDocument()
      expect(getTheme).toHaveBeenCalled()

      // Test dark theme checked disabled
      vi.mocked(getTheme).mockReturnValue('dark')

      const { container: darkContainer } = render(
        <Checkbox checked disabled label="dark-checked-disabled">
          Dark Checked Disabled
        </Checkbox>,
      )

      const darkCheckbox = darkContainer.querySelector('input[type="checkbox"]')
      expect(darkCheckbox).toBeInTheDocument()
      expect(getTheme).toHaveBeenCalled()
    })
  })

  it('should handle hover and active states', () => {
    const { container } = render(
      <Checkbox label="test-checkbox">Interactive Checkbox</Checkbox>,
    )

    const checkbox = container.querySelector('input[type="checkbox"]')
    expect(checkbox).toBeInTheDocument()

    if (checkbox) {
      // Test active state
      fireEvent.mouseDown(checkbox)
      fireEvent.mouseUp(checkbox)

      // Test hover state
      fireEvent.mouseEnter(checkbox)
      fireEvent.mouseLeave(checkbox)
    }
  })

  it('should handle checked state hover', () => {
    const { container } = render(
      <Checkbox checked label="test-checkbox">
        Checked Hover Checkbox
      </Checkbox>,
    )

    const checkbox = container.querySelector('input[type="checkbox"]')
    expect(checkbox).toBeInTheDocument()

    if (checkbox) {
      // Test checked hover state
      fireEvent.mouseEnter(checkbox)
      fireEvent.mouseLeave(checkbox)
    }
  })

  it('should render disabled label text with correct color', () => {
    render(
      <Checkbox disabled label="test-checkbox">
        Disabled Label Text
      </Checkbox>,
    )

    const labelText = screen.getByText('Disabled Label Text')
    expect(labelText).toBeInTheDocument()
  })

  it('should render enabled label text with correct color', () => {
    render(<Checkbox label="test-checkbox">Enabled Label Text</Checkbox>)

    const labelText = screen.getByText('Enabled Label Text')
    expect(labelText).toBeInTheDocument()
  })

  it('should handle all states correctly', () => {
    // Test unchecked state
    const { container: uncheckedContainer } = render(
      <Checkbox label="unchecked">Unchecked Checkbox</Checkbox>,
    )

    const uncheckedCheckbox = uncheckedContainer.querySelector(
      'input[type="checkbox"]',
    )
    expect(uncheckedCheckbox).toBeInTheDocument()
    expect(uncheckedCheckbox).not.toBeChecked()

    // Test checked state
    const { container: checkedContainer } = render(
      <Checkbox checked label="checked">
        Checked Checkbox
      </Checkbox>,
    )

    const checkedCheckbox = checkedContainer.querySelector(
      'input[type="checkbox"]',
    )
    expect(checkedCheckbox).toBeInTheDocument()
    expect(checkedCheckbox).toBeChecked()
  })

  it('should handle disabled states', () => {
    // Disabled unchecked
    const { container: disabledContainer } = render(
      <Checkbox disabled label="disabled">
        Disabled Checkbox
      </Checkbox>,
    )

    const disabledCheckbox = disabledContainer.querySelector(
      'input[type="checkbox"]',
    )
    expect(disabledCheckbox).toBeDisabled()

    // Disabled checked
    const { container: disabledCheckedContainer } = render(
      <Checkbox checked disabled label="disabled-checked">
        Disabled Checked Checkbox
      </Checkbox>,
    )

    const disabledCheckedCheckbox = disabledCheckedContainer.querySelector(
      'input[type="checkbox"]',
    )
    expect(disabledCheckedCheckbox).toBeDisabled()
    expect(disabledCheckedCheckbox).toBeChecked()
  })

  it('should handle interactive states correctly', () => {
    const { container } = render(
      <Checkbox label="interactive">Interactive Checkbox</Checkbox>,
    )

    const checkbox = container.querySelector('input[type="checkbox"]')
    if (checkbox) {
      // Test active state
      fireEvent.mouseDown(checkbox)
      fireEvent.mouseUp(checkbox)

      // Test hover state
      fireEvent.mouseEnter(checkbox)
      fireEvent.mouseLeave(checkbox)
    }
  })

  it('should handle checked hover state correctly', () => {
    const { container } = render(
      <Checkbox checked label="checked-hover">
        Checked Hover Checkbox
      </Checkbox>,
    )

    const checkbox = container.querySelector('input[type="checkbox"]')
    if (checkbox) {
      // Test checked + hover state
      fireEvent.mouseEnter(checkbox)
      fireEvent.mouseLeave(checkbox)
    }
  })

  it('should handle onChange conditional logic', () => {
    const onChange = vi.fn()

    // Test !disabled && onChange?.() - both sides of &&

    // Case 1: disabled=false, onChange exists (should call)
    const { container: enabledContainer } = render(
      <Checkbox label="enabled-onchange" onChange={onChange}>
        Enabled with onChange
      </Checkbox>,
    )

    const enabledCheckbox = enabledContainer.querySelector(
      'input[type="checkbox"]',
    )
    if (enabledCheckbox) {
      fireEvent.click(enabledCheckbox)
    }
    expect(onChange).toHaveBeenCalledWith(true)

    // Case 2: disabled=true, onChange exists (should not call)
    onChange.mockClear()
    const { container: disabledContainer } = render(
      <Checkbox disabled label="disabled-onchange" onChange={onChange}>
        Disabled with onChange
      </Checkbox>,
    )

    const disabledCheckbox = disabledContainer.querySelector(
      'input[type="checkbox"]',
    )
    if (disabledCheckbox) {
      fireEvent.click(disabledCheckbox)
    }
    expect(onChange).not.toHaveBeenCalled()

    // Case 3: disabled=false, no onChange (should not crash)
    const { container: noOnChangeContainer } = render(
      <Checkbox label="no-onchange">No onChange</Checkbox>,
    )

    const noOnChangeCheckbox = noOnChangeContainer.querySelector(
      'input[type="checkbox"]',
    )
    if (noOnChangeCheckbox) {
      fireEvent.click(noOnChangeCheckbox)
    }
    // Should not crash - that's the test
  })
})
