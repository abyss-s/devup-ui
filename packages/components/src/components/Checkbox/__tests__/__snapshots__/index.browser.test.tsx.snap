// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Checkbox > should render basic checkbox 1`] = `
<div>
  <div
    class="display-0-flex--0 align-items-0-center--255 gap-0-8px--255"
  >
    <div
      class="height-0-18px--255 position-0-relative--255 width-0-fit-content--255"
    >
      <input
        class="background-0-light-dark(var(--primary,#6159D4),var(--primary,#6670F9))-6529705465913413318-1 border-0-none-6529705465913413318-1 background-0-light-dark(#F0F0F3,#47474A)-9407739846625397859-1 background-0-light-dark(#F0F0F3,#47474A)-14172363753176421546-1 accent-color-0-light-dark(var(--primary,#6159D4),var(--primary,#6670F9))--1 appearance-0-none--1 background-0-light-dark(#FFF,var(--inputBg,#2E2E2E))--1 border-0-1px solid light-dark(var(--border,#E0E0E0),var(--border,#333))--1 border-radius-0-2px--1 height-0-16px--1 width-0-16px--1 margin-0-0--1 background-0-light-dark(color-mix(in srgb,var(--primary,#6159D4) 20%,#FFF 80%),color-mix(in srgb,var(--primary,#6670F9) 30%,#000 70%))-15425828959012638752-1 background-0-light-dark(color-mix(in srgb,var(--primary,#6159D4) 100%,#000 15%),color-mix(in srgb,var(--primary,#6670F9) 100%,#FFF 15%))-15130211206989154723-1 background-0-light-dark(color-mix(in srgb,var(--primary,#6159D4) 10%,#FFF 90%),color-mix(in srgb,var(--primary,#6670F9) 20%,#000 80%))-8380715471663921674-1 border-0-1px solid light-dark(var(--primary,#6159D4),var(--primary,#6670F9))-8380715471663921674-1 cursor-0-pointer--1 "
        id="«r0»"
        type="checkbox"
      />
    </div>
    <label
      class="cursor-0-pointer--255"
      for="«r0»"
    >
      <span
        class="font-size-0-14px--255 user-select-0-none--255 color-0-light-dark(var(--text,#2F2F2F),var(--text,#EDEDED))--255"
      >
        Test Checkbox
      </span>
    </label>
  </div>
</div>
`;

exports[`Checkbox > should render checkbox with custom child 1`] = `
<div>
  <div
    class="display-0-flex--0 align-items-0-center--255 gap-0-8px--255"
  >
    <div
      class="height-0-18px--255 position-0-relative--255 width-0-fit-content--255"
    >
      <input
        class="background-0-light-dark(var(--primary,#6159D4),var(--primary,#6670F9))-6529705465913413318-1 border-0-none-6529705465913413318-1 background-0-light-dark(#F0F0F3,#47474A)-9407739846625397859-1 background-0-light-dark(#F0F0F3,#47474A)-14172363753176421546-1 accent-color-0-light-dark(var(--primary,#6159D4),var(--primary,#6670F9))--1 appearance-0-none--1 background-0-light-dark(#FFF,var(--inputBg,#2E2E2E))--1 border-0-1px solid light-dark(var(--border,#E0E0E0),var(--border,#333))--1 border-radius-0-2px--1 height-0-16px--1 width-0-16px--1 margin-0-0--1 background-0-light-dark(color-mix(in srgb,var(--primary,#6159D4) 20%,#FFF 80%),color-mix(in srgb,var(--primary,#6670F9) 30%,#000 70%))-15425828959012638752-1 background-0-light-dark(color-mix(in srgb,var(--primary,#6159D4) 100%,#000 15%),color-mix(in srgb,var(--primary,#6670F9) 100%,#FFF 15%))-15130211206989154723-1 background-0-light-dark(color-mix(in srgb,var(--primary,#6159D4) 10%,#FFF 90%),color-mix(in srgb,var(--primary,#6670F9) 20%,#000 80%))-8380715471663921674-1 border-0-1px solid light-dark(var(--primary,#6159D4),var(--primary,#6670F9))-8380715471663921674-1 cursor-0-pointer--1 "
        id="«r4»"
        type="checkbox"
      />
    </div>
    <label
      class="cursor-0-pointer--255"
      for="«r4»"
    >
      <div>
        Custom Child
      </div>
    </label>
  </div>
</div>
`;

exports[`Checkbox > should render checked checkbox 1`] = `
<div>
  <div
    class="display-0-flex--0 align-items-0-center--255 gap-0-8px--255"
  >
    <div
      class="height-0-18px--255 position-0-relative--255 width-0-fit-content--255"
    >
      <input
        checked=""
        class="background-0-light-dark(var(--primary,#6159D4),var(--primary,#6670F9))-6529705465913413318-1 border-0-none-6529705465913413318-1 background-0-light-dark(#F0F0F3,#47474A)-9407739846625397859-1 background-0-light-dark(#F0F0F3,#47474A)-14172363753176421546-1 accent-color-0-light-dark(var(--primary,#6159D4),var(--primary,#6670F9))--1 appearance-0-none--1 background-0-light-dark(#FFF,var(--inputBg,#2E2E2E))--1 border-0-1px solid light-dark(var(--border,#E0E0E0),var(--border,#333))--1 border-radius-0-2px--1 height-0-16px--1 width-0-16px--1 margin-0-0--1 background-0-light-dark(color-mix(in srgb,var(--primary,#6159D4) 20%,#FFF 80%),color-mix(in srgb,var(--primary,#6670F9) 30%,#000 70%))-15425828959012638752-1 background-0-light-dark(color-mix(in srgb,var(--primary,#6159D4) 100%,#000 15%),color-mix(in srgb,var(--primary,#6670F9) 100%,#FFF 15%))-15130211206989154723-1 background-0-light-dark(color-mix(in srgb,var(--primary,#6159D4) 10%,#FFF 90%),color-mix(in srgb,var(--primary,#6670F9) 20%,#000 80%))-8380715471663921674-1 border-0-1px solid light-dark(var(--primary,#6159D4),var(--primary,#6670F9))-8380715471663921674-1 cursor-0-pointer--1 "
        id="«r1»"
        type="checkbox"
      />
      <svg
        class="left-0-50%--255 pointer-events-0-none--255 position-0-absolute--255 top-0-8px--255 transform-0-translate(-50%,-50%)--255"
        height="10"
        viewBox="0 0 12 10"
        width="12"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M11.6474 0.807113C12.0992 1.23373 12.1195 1.94575 11.6929 2.39745L5.31789 9.14745C5.10214 9.37589 4.80069 9.50369 4.48649 9.49992C4.1723 9.49615 3.874 9.36114 3.6638 9.12759L0.288803 5.37759C-0.126837 4.91576 -0.0893993 4.20444 0.372424 3.7888C0.834247 3.37315 1.54557 3.41059 1.96121 3.87242L4.51994 6.71544L10.0571 0.852551C10.4837 0.400843 11.1957 0.3805 11.6474 0.807113Z"
          fill="#FFF"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <label
      class="cursor-0-pointer--255"
      for="«r1»"
    >
      <span
        class="font-size-0-14px--255 user-select-0-none--255 color-0-light-dark(var(--text,#2F2F2F),var(--text,#EDEDED))--255"
      >
        Test Checkbox
      </span>
    </label>
  </div>
</div>
`;

exports[`Checkbox > should render checked checkbox with custom colors 1`] = `
<div>
  <div
    class="display-0-flex--0 align-items-0-center--255 gap-0-8px--255"
  >
    <div
      class="height-0-18px--255 position-0-relative--255 width-0-fit-content--255"
    >
      <input
        checked=""
        class="background-0-light-dark(var(--primary,#6159D4),var(--primary,#6670F9))-6529705465913413318-1 border-0-none-6529705465913413318-1 background-0-light-dark(#F0F0F3,#47474A)-9407739846625397859-1 background-0-light-dark(#F0F0F3,#47474A)-14172363753176421546-1 accent-color-0-light-dark(var(--primary,#6159D4),var(--primary,#6670F9))--1 appearance-0-none--1 background-0-light-dark(#FFF,var(--inputBg,#2E2E2E))--1 border-0-1px solid light-dark(var(--border,#E0E0E0),var(--border,#333))--1 border-radius-0-2px--1 height-0-16px--1 width-0-16px--1 margin-0-0--1 background-0-light-dark(color-mix(in srgb,var(--primary,#6159D4) 20%,#FFF 80%),color-mix(in srgb,var(--primary,#6670F9) 30%,#000 70%))-15425828959012638752-1 background-0-light-dark(color-mix(in srgb,var(--primary,#6159D4) 100%,#000 15%),color-mix(in srgb,var(--primary,#6670F9) 100%,#FFF 15%))-15130211206989154723-1 background-0-light-dark(color-mix(in srgb,var(--primary,#6159D4) 10%,#FFF 90%),color-mix(in srgb,var(--primary,#6670F9) 20%,#000 80%))-8380715471663921674-1 border-0-1px solid light-dark(var(--primary,#6159D4),var(--primary,#6670F9))-8380715471663921674-1 cursor-0-pointer--1 "
        id="«rj»"
        style="--primary: #ff0000; --border: #00ff00; --text: #0000ff; --inputBg: #ffff00;"
        type="checkbox"
      />
      <svg
        class="left-0-50%--255 pointer-events-0-none--255 position-0-absolute--255 top-0-8px--255 transform-0-translate(-50%,-50%)--255"
        height="10"
        viewBox="0 0 12 10"
        width="12"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M11.6474 0.807113C12.0992 1.23373 12.1195 1.94575 11.6929 2.39745L5.31789 9.14745C5.10214 9.37589 4.80069 9.50369 4.48649 9.49992C4.1723 9.49615 3.874 9.36114 3.6638 9.12759L0.288803 5.37759C-0.126837 4.91576 -0.0893993 4.20444 0.372424 3.7888C0.834247 3.37315 1.54557 3.41059 1.96121 3.87242L4.51994 6.71544L10.0571 0.852551C10.4837 0.400843 11.1957 0.3805 11.6474 0.807113Z"
          fill="#FFF"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <label
      class="cursor-0-pointer--255"
      for="«rj»"
    >
      <span
        class="font-size-0-14px--255 user-select-0-none--255 color-0-light-dark(var(--text,#2F2F2F),var(--text,#EDEDED))--255"
      >
        Test Checkbox
      </span>
    </label>
  </div>
</div>
`;

exports[`Checkbox > should render disabled and checked checkbox 1`] = `
<div>
  <div
    class="display-0-flex--0 align-items-0-center--255 gap-0-8px--255"
  >
    <div
      class="height-0-18px--255 position-0-relative--255 width-0-fit-content--255"
    >
      <input
        checked=""
        class="background-0-light-dark(var(--primary,#6159D4),var(--primary,#6670F9))-6529705465913413318-1 border-0-none-6529705465913413318-1 background-0-light-dark(#F0F0F3,#47474A)-9407739846625397859-1 background-0-light-dark(#F0F0F3,#47474A)-14172363753176421546-1 accent-color-0-light-dark(var(--primary,#6159D4),var(--primary,#6670F9))--1 appearance-0-none--1 background-0-light-dark(#FFF,var(--inputBg,#2E2E2E))--1 border-0-1px solid light-dark(var(--border,#E0E0E0),var(--border,#333))--1 border-radius-0-2px--1 height-0-16px--1 width-0-16px--1 margin-0-0--1    cursor-0-not-allowed--1 "
        disabled=""
        id="«r3»"
        type="checkbox"
      />
      <svg
        class="left-0-50%--255 pointer-events-0-none--255 position-0-absolute--255 top-0-8px--255 transform-0-translate(-50%,-50%)--255"
        height="10"
        viewBox="0 0 12 10"
        width="12"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M11.6474 0.807113C12.0992 1.23373 12.1195 1.94575 11.6929 2.39745L5.31789 9.14745C5.10214 9.37589 4.80069 9.50369 4.48649 9.49992C4.1723 9.49615 3.874 9.36114 3.6638 9.12759L0.288803 5.37759C-0.126837 4.91576 -0.0893993 4.20444 0.372424 3.7888C0.834247 3.37315 1.54557 3.41059 1.96121 3.87242L4.51994 6.71544L10.0571 0.852551C10.4837 0.400843 11.1957 0.3805 11.6474 0.807113Z"
          fill="light-dark(#D6D7DE, #373737)"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <label
      class="cursor-0-not-allowed--255"
      for="«r3»"
    >
      <span
        class="font-size-0-14px--255 user-select-0-none--255 color-0-light-dark(#D6D7DE,#373737)--255"
      >
        Test Checkbox
      </span>
    </label>
  </div>
</div>
`;

exports[`Checkbox > should render disabled checkbox 1`] = `
<div>
  <div
    class="display-0-flex--0 align-items-0-center--255 gap-0-8px--255"
  >
    <div
      class="height-0-18px--255 position-0-relative--255 width-0-fit-content--255"
    >
      <input
        class="background-0-light-dark(var(--primary,#6159D4),var(--primary,#6670F9))-6529705465913413318-1 border-0-none-6529705465913413318-1 background-0-light-dark(#F0F0F3,#47474A)-9407739846625397859-1 background-0-light-dark(#F0F0F3,#47474A)-14172363753176421546-1 accent-color-0-light-dark(var(--primary,#6159D4),var(--primary,#6670F9))--1 appearance-0-none--1 background-0-light-dark(#FFF,var(--inputBg,#2E2E2E))--1 border-0-1px solid light-dark(var(--border,#E0E0E0),var(--border,#333))--1 border-radius-0-2px--1 height-0-16px--1 width-0-16px--1 margin-0-0--1    cursor-0-not-allowed--1 "
        disabled=""
        id="«r2»"
        type="checkbox"
      />
    </div>
    <label
      class="cursor-0-not-allowed--255"
      for="«r2»"
    >
      <span
        class="font-size-0-14px--255 user-select-0-none--255 color-0-light-dark(#D6D7DE,#373737)--255"
      >
        Test Checkbox
      </span>
    </label>
  </div>
</div>
`;

exports[`Checkbox > should render disabled checkbox with custom colors 1`] = `
<div>
  <div
    class="display-0-flex--0 align-items-0-center--255 gap-0-8px--255"
  >
    <div
      class="height-0-18px--255 position-0-relative--255 width-0-fit-content--255"
    >
      <input
        class="background-0-light-dark(var(--primary,#6159D4),var(--primary,#6670F9))-6529705465913413318-1 border-0-none-6529705465913413318-1 background-0-light-dark(#F0F0F3,#47474A)-9407739846625397859-1 background-0-light-dark(#F0F0F3,#47474A)-14172363753176421546-1 accent-color-0-light-dark(var(--primary,#6159D4),var(--primary,#6670F9))--1 appearance-0-none--1 background-0-light-dark(#FFF,var(--inputBg,#2E2E2E))--1 border-0-1px solid light-dark(var(--border,#E0E0E0),var(--border,#333))--1 border-radius-0-2px--1 height-0-16px--1 width-0-16px--1 margin-0-0--1    cursor-0-not-allowed--1 "
        disabled=""
        id="«rk»"
        style="--primary: #ff0000; --border: #00ff00; --text: #0000ff; --inputBg: #ffff00;"
        type="checkbox"
      />
    </div>
    <label
      class="cursor-0-not-allowed--255"
      for="«rk»"
    >
      <span
        class="font-size-0-14px--255 user-select-0-none--255 color-0-light-dark(#D6D7DE,#373737)--255"
      >
        Test Checkbox
      </span>
    </label>
  </div>
</div>
`;

exports[`Checkbox > should render with custom colors 1`] = `
<div>
  <div
    class="display-0-flex--0 align-items-0-center--255 gap-0-8px--255"
  >
    <div
      class="height-0-18px--255 position-0-relative--255 width-0-fit-content--255"
    >
      <input
        class="background-0-light-dark(var(--primary,#6159D4),var(--primary,#6670F9))-6529705465913413318-1 border-0-none-6529705465913413318-1 background-0-light-dark(#F0F0F3,#47474A)-9407739846625397859-1 background-0-light-dark(#F0F0F3,#47474A)-14172363753176421546-1 accent-color-0-light-dark(var(--primary,#6159D4),var(--primary,#6670F9))--1 appearance-0-none--1 background-0-light-dark(#FFF,var(--inputBg,#2E2E2E))--1 border-0-1px solid light-dark(var(--border,#E0E0E0),var(--border,#333))--1 border-radius-0-2px--1 height-0-16px--1 width-0-16px--1 margin-0-0--1 background-0-light-dark(color-mix(in srgb,var(--primary,#6159D4) 20%,#FFF 80%),color-mix(in srgb,var(--primary,#6670F9) 30%,#000 70%))-15425828959012638752-1 background-0-light-dark(color-mix(in srgb,var(--primary,#6159D4) 100%,#000 15%),color-mix(in srgb,var(--primary,#6670F9) 100%,#FFF 15%))-15130211206989154723-1 background-0-light-dark(color-mix(in srgb,var(--primary,#6159D4) 10%,#FFF 90%),color-mix(in srgb,var(--primary,#6670F9) 20%,#000 80%))-8380715471663921674-1 border-0-1px solid light-dark(var(--primary,#6159D4),var(--primary,#6670F9))-8380715471663921674-1 cursor-0-pointer--1 "
        id="«rh»"
        style="--primary: #ff0000; --border: #00ff00; --text: #0000ff; --inputBg: #ffff00;"
        type="checkbox"
      />
    </div>
    <label
      class="cursor-0-pointer--255"
      for="«rh»"
    >
      <span
        class="font-size-0-14px--255 user-select-0-none--255 color-0-light-dark(var(--text,#2F2F2F),var(--text,#EDEDED))--255"
      >
        Test Checkbox
      </span>
    </label>
  </div>
</div>
`;

exports[`Checkbox > should render with partial custom colors 1`] = `
<div>
  <div
    class="display-0-flex--0 align-items-0-center--255 gap-0-8px--255"
  >
    <div
      class="height-0-18px--255 position-0-relative--255 width-0-fit-content--255"
    >
      <input
        class="background-0-light-dark(var(--primary,#6159D4),var(--primary,#6670F9))-6529705465913413318-1 border-0-none-6529705465913413318-1 background-0-light-dark(#F0F0F3,#47474A)-9407739846625397859-1 background-0-light-dark(#F0F0F3,#47474A)-14172363753176421546-1 accent-color-0-light-dark(var(--primary,#6159D4),var(--primary,#6670F9))--1 appearance-0-none--1 background-0-light-dark(#FFF,var(--inputBg,#2E2E2E))--1 border-0-1px solid light-dark(var(--border,#E0E0E0),var(--border,#333))--1 border-radius-0-2px--1 height-0-16px--1 width-0-16px--1 margin-0-0--1 background-0-light-dark(color-mix(in srgb,var(--primary,#6159D4) 20%,#FFF 80%),color-mix(in srgb,var(--primary,#6670F9) 30%,#000 70%))-15425828959012638752-1 background-0-light-dark(color-mix(in srgb,var(--primary,#6159D4) 100%,#000 15%),color-mix(in srgb,var(--primary,#6670F9) 100%,#FFF 15%))-15130211206989154723-1 background-0-light-dark(color-mix(in srgb,var(--primary,#6159D4) 10%,#FFF 90%),color-mix(in srgb,var(--primary,#6670F9) 20%,#000 80%))-8380715471663921674-1 border-0-1px solid light-dark(var(--primary,#6159D4),var(--primary,#6670F9))-8380715471663921674-1 cursor-0-pointer--1 "
        id="«ri»"
        style="--primary: #ff0000; --text: #0000ff;"
        type="checkbox"
      />
    </div>
    <label
      class="cursor-0-pointer--255"
      for="«ri»"
    >
      <span
        class="font-size-0-14px--255 user-select-0-none--255 color-0-light-dark(var(--text,#2F2F2F),var(--text,#EDEDED))--255"
      >
        Test Checkbox
      </span>
    </label>
  </div>
</div>
`;
